src/management-system-v2/lib/useColumnWidth.tsx

```diff
@@ -1,206 +1,258 @@
-    (event: MouseEvent) => {
-      if (resizingColumn.current) {
-        /* Calculate new Width */
-        const currentWidth = columnRef.current?.offsetWidth || 0;
-        const newWidth = currentWidth + event.clientX - startX.current;
+    onlySelectedColumnsCauseRerender.current =
+      true; /* This is only true once the columns are set to the saved values (after hydration) */
+  }, [hydrated, columns, columnsInPreferences]);
 
-        /* Update UI */
-        setColumnWidths((old) => ({ ...old, [resizingColumn.current]: newWidth }));
+  const handleResize = useCallback(
+    (index: number) =>
+      (e: any, { size }: any) => {
+        let thWidth = size.width;
+        /* Antdesign can handle a size of 'auto' */
+        /* However, react-resizeable can't */
+        if (Number.isNaN(thWidth) || thWidth === null) {
+          /* In this case we need to read the actual width from the DOM */
+          const th = e.target.closest('th');
+          if (!th) return; // This should never be the case
+          thWidth = th.getBoundingClientRect().width;
+        }
+        const newWidth = Math.max(thWidth, minWidth);
 
-        console.log('resizingColumn.current', resizingColumn.current, newWidth);
+        /* Change state */
+        /* Note: changing the preferences should suffice (if it triggers a state change aswell), however, changing the state here seems to be smoother */
+        /* Therefore only column selection chaneges in the localstorage are changing the state (after hydration) */
+        setResizeableColumns((old) => {
+          const newWidths = [...old];
+          // @ts-ignore
+          newWidths[index] = { ...newWidths[index], width: newWidth };
+          return newWidths;
+        });
 
         /* Update Preferences */
         const newColumnsInPreferences = columnsInPreferences.map((column: any) => {
-          if (column.name === resizingColumn.current) {
+          if (column.name === columns[index].title) {
             return { name: column.name, width: newWidth };
           }
           return column;
         });
         addPreferences({ [preferenceKey]: newColumnsInPreferences });
+      },
+    [minWidth, columnsInPreferences, addPreferences, preferenceKey, columns],
+  );
 
-        /* Remove the indicator div */
-        if (indicatorRef.current) {
-          document.body.removeChild(indicatorRef.current);
-          indicatorRef.current = null;
-        }
+  const columsWithResize = resizeableColumns.map((column: any, index: number) => {
+    if (notResizeabel.includes(column.key)) return column;
+
+    return {
+      ...column,
+      onHeaderCell: (column: any) => ({
+        width: column.width,
+        onResize: handleResize(index),
+      }),
+    };
+  }) as TableColumnsType<any>;
+
+  return useTruncateColumnText(columsWithResize);
+  // return columsWithResize;
+};
+
+export const ResizableTitle = (props: any) => {
+  const { onResize, width, ...restProps } = props;
```

Comment:
Why no destructuring directly in the function declaration? And is it possible to narrow the type down?

Reply:
Made suggested changes